bin/menu/main_menu.dart

```diff
@@ -0,0 +1,120 @@
+import '../collection/node.dart';
+import '../utils/const.dart';
+import '../utils/logger.dart';
+import '../utils/store.dart';
+import 'base/choice.dart';
+import 'base/entry.dart';
+import 'base/menu.dart';
+
+class MainMenu extends Choice {
+  MainMenu() : super('') {
+    addMenu(_NodeOperationMenu(
+        name: "Get the immediate parents of a node",
+        operation: (Node n) => _printNodes("Immediate parents", n.parent, n.nodeId)));
+    addMenu(_NodeOperationMenu(
+        name: "Get the immediate children of a node",
+        operation: (Node n) => _printNodes("Immediate children", n.children, n.nodeId)));
+    addMenu(_NodeOperationMenu(
+        name: "Get the ancestors of a node",
+        operation: (Node n) => _printNodes("Ancestors", n.getAncestorNodes(), n.nodeId)));
+    addMenu(_NodeOperationMenu(
+        name: "Get the descendants of a node",
+        operation: (Node n) => _printNodes("Descendants", n.getDescendantNodes(), n.nodeId)));
+    addMenu(_ParentChildOperationMenu(
+        name: "Delete dependency from a tree",
+        operation: ({required int parentId, required int childId}) {
+          Store.instance.graph.removeDepedency(parentId: parentId, childId: childId);
+          Log.info(
+              "Removed dependency from parent $parentId to child $childId");
+        }));
+    addMenu(_NodeOperationMenu(
+        name: "Delete a node from a tree",
+        operation: (Node n) {
+          Store.instance.graph.deleteNode(n.nodeId);
+          Log.info("Deleted node with id ${n.nodeId}");
+        }));
+    addMenu(_ParentChildOperationMenu(
+        name: "Add a new dependency to a tree",
+        operation: ({required int parentId, required int childId}) {
+          Store.instance.graph.addDepedency(parentId: parentId, childId: childId);
+          Log.info(
+              "Added a dependency from parent $parentId to child $childId");
+        }));
+    addMenu(_AddNodeMenu());
+    addMenu(_ExitMenu());
```

Comment:
Why in constructor ? 
Constructor should have minor pre-processing data before usage 
This whole thing is a create function 
Can be static or can be a member function
Don't do it inside constructor 

Modularize better 
Have a list passed to this function or create a list in the same class containing the menu objects
Inside createMenu() just loop through the menus list and do addMenu(menus[index]).

Easier to add another menu in the future by just finding the menu list and adding one